rules/manager.go

```diff
@@ -267,22 +268,34 @@ type Group struct {
 
 	metrics *Metrics
 
+	// This function will be used before each rule group evaluation if not nil.
+	// Deprecated: Superceded by evalIterationFunc which enables arbitrary instrumentation around the default rule group evaluation iteration
 	ruleGroupPostProcessFunc RuleGroupPostProcessFunc
+
+	// rule group evaluation iteration function, which by default invokes DefaultEvalIterationFunc
+	evalIterationFunc GroupEvalIterationFunc
 }
 
-// This function will be used before each rule group evaluation if not nil.
-// Use this function type if the rule group post processing is needed.
+// Use this function type if rule group pre-processing is needed.
+// Deprecated:  Succeeded by GroupEvalIterationFunc
```

Comment:
Is it reasonable to deprecate this and still use it in the API to `Update()`?
Everyone using it will get a linter warning for the deprecation, and a compile error to add an extra parameter. Maybe just force them to use the new style?

Reply:
I don't mind simply replacing it.  My team is the one who introduced the old callback, and we're the ones who who will be moving to new callback.

All consumers supplying ```nil``` to ```Update()``` should be unaffected when they rebuild, which is probably (hopefully) the default case.

I'll post a new revision today with the replacement.